crates/derive/src/stages/l1_retrieval.rs

```diff
@@ -60,7 +60,13 @@ where
             );
         }
 
-        let data = self.data.as_mut().expect("Cannot be None").next();
+        let data = self
+            .data
+            .as_mut()
+            .expect("Cannot be None")
+            .next()
+            .await
+            .unwrap_or(Err(StageError::Empty));
```

Comment:
This `unwrap_or` looks in an odd spot - if there's an error, we want to catch the EOF in the below match. What is the significance of `StageError::Empty`?

Reply:
Ah this is an unwrap on the `Option<>`. I've modified it to unwrap to `StageError::Eof`... don't think `StageError::Empty` makes sense .